tests/com/twitter/intellij/pants/util/PantsUtilTest.java

```diff
@@ -36,4 +38,40 @@ public void run() {
     // Make sure they are identical, meaning that no new JDK was created on the 2nd find.
     assertTrue(sdkA.get() == sdkB.get());
   }
+
+  public void testisBUILDFilePath() {
+    assertFalse("pants.ini file should not be interpreted as a BUILD file",
+                PantsUtil.isBUILDFilePath(pantsIniFilePath));
+
+    assertFalse("made up file path should not be interpreted as a BUILD file",
+                PantsUtil.isBUILDFilePath(nonexistentFilePath));
+
+    assertTrue("made up BUILD file path should be interpreted as a BUILD file path",
+               PantsUtil.isBUILDFilePath(nonexistentBuildFilePath));
+
+    assertTrue("path to invalid, existing BUILD file should be interpreted as a BUILD file path",
+               PantsUtil.isBUILDFilePath(invalidBuildFilePath));
+  }
+
+  public void testListAllTargets() {
+    assertEquals("pants.ini file should have no targets",
+                 PantsUtil.listAllTargets(pantsIniFilePath),
+                 Lists.newArrayList());
+
+    assertEquals("made up file path should have no targets",
+                 PantsUtil.listAllTargets(nonexistentFilePath),
+                 Lists.newArrayList());
+
+    try {
+      PantsUtil.listAllTargets(nonexistentBuildFilePath);
+      fail(String.format("%s should have been thrown", PantsException.class));
+    } catch (PantsException ignored) {
+    }
+
+    try {
+      PantsUtil.listAllTargets(invalidBuildFilePath);
+      fail(String.format("%s should have been thrown"));
```

Comment:
is there a particular exception to be caught? what is %s for?

Reply:
Whoops. I used `String.format()` following the example you gave in your previous comment, but forgot that means I need to check it. Just pushed a change fixing this.